API/NoDocs/Microsoft.Extensions.AsyncState/IAsyncContext.cs

```diff
@@ -0,0 +1,12 @@
+// Assembly 'Microsoft.Extensions.AsyncState'
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.Extensions.AsyncState;
+
+public interface IAsyncContext<T> where T : notnull
```

Comment:
Were these (AsyncState, IAsyncContext) considered for moving into the platform as concepts?

Assuming they belong up here, they seem to be burning relatively general purpose names. We try to keep type names in the platform unique across all namespaces, more or less.

cc @stephentoub

Reply:
> Were these (AsyncState, IAsyncContext) considered for moving into the platform as concepts?

They're specifically about DI and from my perspective don't make sense lower in the stack.  Which Microsoft.Extensions library they end up living in I don't think matters much, and repo doesn't really matter, so having them in a Microsoft.Extensions.AsyncState library seems reasonable to me.

Reply:
I agree,